app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::BaseController < ApplicationController
+
+  before_action :require_admin
+
+  def require_admin
+    render file: "/public/404" unless current_user && current_user.admin?
```

Comment:
@Dpalazzari @njgheorghita Do we have any specific message on the 404 page to let the user know what happened? Or is it intentional not to even give the impression that the page exists for some users? I could see the benefit, just wanted to confirm the thinking behind this implementation. The alternative I have in mind is flash message and redirect to `root_path`. FYI @drod1000